lib/clamp-link.in

```diff
@@ -206,7 +206,7 @@ _thinlto_path() {
 
     # augment arguments to clang-offload-bundler
     CLANG_OFFLOAD_BUNDLER_INPUTS_ARGS+=",$KERNEL-$AMDGPU_TARGET.hsaco"
-    CLANG_OFFLOAD_BUNDLER_TARGETS_ARGS+=",hcc-amdgcn--amdhsa-$AMDGPU_TARGET"
+    CLANG_OFFLOAD_BUNDLER_TARGETS_ARGS+=",hcc-amdgcn-amd-amdhsa--$AMDGPU_TARGET"
```

Comment:
I'm not sure if removing hcc- will work. It will break this causing an error:
```
[100%] Linking CXX executable ../bin/cmake-test
error: invalid target 'amdgcn-amd-amdhsa--gfx803', unknown offloading kind 'amdgcn', unknown target triple 'amd-amdhsa--gfx803'.
error: invalid target 'amdgcn-amd-amdhsa--gfx900', unknown offloading kind 'amdgcn', unknown target triple 'amd-amdhsa--gfx900'.
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
cmake-tests/CMakeFiles/cmake-test.dir/build.make:97: recipe for target 'bin/cmake-test' failed
```
This happens here where we need to specify the kind before the target triple:
https://github.com/RadeonOpenCompute/hcc-clang-upgrade/blob/fa45c74b2046b68a7933531f3ec35858b79448c0/tools/clang-offload-bundler/ClangOffloadBundler.cpp#L976

Reply:
That looks like a test. I suspect that the tests need to be updated to use the new way to specify a target. The idea is that the bundled code object uses the standard target identification specified in https://llvm.org/docs/AMDGPUUsage.html#code-object-target-identification .

Reply:
In the clang offload bundler has the format `<offload kind>-<triple>` and therefore the `hcc-` is needed, see

https://github.com/RadeonOpenCompute/hcc-clang-upgrade/blob/fa45c74b2046b68a7933531f3ec35858b79448c0/tools/clang-offload-bundler/ClangOffloadBundler.cpp#L114

Reply:
OK. So do not change this. Update the triple_to_hsa_isa() to check the offload kind.

Still need to add the +xnack if needed.

Reply:
At the moment we don't support either page migration or Carrizo so do you agree that we don't need to address +xnack in this PR?

Reply:
Are we supporting Ryzen with Vega APU?

Reply:
not at point